### PR TITLE
Script logic

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✔                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: src/publishCommand/publishManager.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        links: this.linkBuilder.createLinks({
+          layerName: unifiedLayerName,
+          serverUrl: publicMapServerUrl,
+        }),
+      });
+    }
+  }
+
+  private async validateRunConditions(metadata: LayerMetadata): Promise<void> {
+    //await this.metadataValidator.validate(metadata);
```
Comment (asafMasa) [inline src/publishCommand/publishManager.ts:144]: remarks

File: tests/integration/publishCommand/publishCommand.spec.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+import { catalogMock, catalogPublishMock } from '../../mock/clients/catalogClient';
+import { PublishCommandCliTrigger } from './helpers/CliTrigger';
+
+describe('PublishCommand', function () {
+  let cli: PublishCommandCliTrigger;
+  // let consoleLogMock: jest.SpyInstance;
+  let processExitMock: jest.SpyInstance;
+
+  beforeEach(function () {
+    // consoleLogMock = jest.spyOn(global.console, 'log');
```
Comment (asafMasa) [inline tests/integration/publishCommand/publishCommand.spec.ts:18]: remarks

File: tests/integration/publishCommand/publishCommand.spec.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+import { SERVICES } from '../../../src/common/constants';
+import { MapPublisherClient } from '../../../src/clients/mapPublisherClient';
+import { CatalogClient } from '../../../src/clients/catalogClient';
+import { mapPublisherMock, mapPublishLayerMock } from '../../mock/clients/mapPublisherClient';
+import { catalogMock, catalogPublishMock } from '../../mock/clients/catalogClient';
+import { PublishCommandCliTrigger } from './helpers/CliTrigger';
+
+describe('PublishCommand', function () {
+  let cli: PublishCommandCliTrigger;
+  // let consoleLogMock: jest.SpyInstance;
```
Comment (asafMasa) [inline tests/integration/publishCommand/publishCommand.spec.ts:14]: remarks

File: tests/integration/publishCommand/publishCommand.spec.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+  let processExitMock: jest.SpyInstance;
+
+  beforeEach(function () {
+    // consoleLogMock = jest.spyOn(global.console, 'log');
+    // consoleLogMock.mockReturnValue(undefined);
+    // jest.spyOn(global.console, 'error').mockReturnValue(undefined); // prevent cli error logs from messing with test log on bad path tests
+    processExitMock = jest.spyOn(global.process, 'exit');
+    processExitMock.mockReturnValueOnce(undefined); //prevent cli exit from killing the test
+
+    //overriding  containers that are registered with decorators dont work with current implementation
```
Comment (asafMasa) [inline tests/integration/publishCommand/publishCommand.spec.ts:24]: remarks

**Review by asafMasa — CHANGES_REQUESTED** [review]
remove helm files
throw error on update scenario "Not Supported"
basic readme about with the link to excel parameters

**Review by asafMasa — APPROVED** [review]

**Review by asafMasa — APPROVED** [review]

**Review by AnguIar — APPROVED** [review]

**Review by AnguIar — APPROVED** [review]

**Comment by AnguIar** [comment]
בס"ד
ישר כוח

**Pull request merged**